packages/manager/modules/sms/src/sms/sms/outgoing/telecom-sms-sms-outgoing.controller.js

```diff
@@ -125,83 +123,75 @@ export default class {
   }
 
   /**
-   * Get details.
-   * @param  {Object} id
+   * Get Ptts if !== 1.
+   * @param  {Object} sms
    * @return {Promise}
    */
-  getDetails(id) {
+  getPtts(sms) {
     this.outgoing.isLoading = true;
-    return this.api.smsOutgoing
-      .get({
-        serviceName: this.$stateParams.serviceName,
-        id,
+    return this.api.smsJobs
+      .getPtts({
+        ptt: sms.ptt,
       })
-      .$promise.then((outgoing) => {
-        if (isEmpty(outgoing.sender)) {
-          set(outgoing, 'isShortNumber', true);
-          set(
-            outgoing,
-            'sender',
-            this.$translate.instant('sms_sms_outgoing_number_allowed_response'),
-          );
-        }
-        return this.api.smsJobs
-          .getPtts({
-            ptt: outgoing.ptt,
-          })
-          .$promise.then((ptt) => {
-            set(outgoing, 'status', ptt.comment);
-            return outgoing;
-          });
+      .$promise.then((ptt) => {
+        set(sms, 'status', ptt.comment);
+        return sms;
       });
   }
 
-  /**
-   * Toggle show filter.
-   */
-  toggleShowFilter() {
-    this.outgoing.showFilter = !this.outgoing.showFilter;
-    this.outgoing.filterBy = {
-      receiver: undefined,
-      sender: undefined,
-    };
-    if (this.outgoing.showFilter === false) {
-      this.refresh();
-    }
+  onRowSelect($row, $rows) {
+    this.outgoing.selected = $rows;
```

Comment:
**question:** Why are you setting all the rows as selected?

Reply:
I need for the massive (selective) deletion.